File: PHP/Easy/Filters/Hash_Table/1399__Count_Largest_Group.php
<?php
function countLargestGroup($n)
{
    $groups = [];
    for ($num = 1; $num <= $n; $num++) {
        $digit_sum = array_sum(str_split((string)$num));
        $groups[$digit_sum] = ($groups[$digit_sum] ?? 0) + 1;
    }
    $max_size = max($groups);
    $count = count(array_filter($groups, fn($size) => $size == $max_size));

    return $count;
}

$n = 13;
echo countLargestGroup($n);

File: Python/Easy/Filter/Hash_Table/1399__Count_Largest_Group.py
def countLargestGroup(n: int) -> int:
    groups = {}
    for num in range(1, n + 1):
        digit_sum = sum(int(d) for d in str(num))
        groups[digit_sum] = groups.get(digit_sum, 0) + 1

    max_size = max(groups.values())
    count = sum(1 for size in groups.values() if size == max_size)

    return count


n = 46
print(countLargestGroup(n))
